manifest-build-tools/application/update_rackhd_version.py

```diff
@@ -0,0 +1,207 @@
+        """
+        Parse debian control file
+        :param control_path: the path of control file
+        :return: a dictionay which contains all the package in field Depends
+        """
+        if not os.path.isfile(control_path):
+            raise RuntimeError("Can't parse {0} because it is not a file".format(control))
+
+        for paragraph in deb822.Deb822.iter_paragraphs(open(control_path)):
+            for item in paragraph.items():
+                if item[0] == 'Depends':
+                    packages = item[1].split("\n")
+                    return packages
+        return None
+
+    def _update_dependency(self, debian_dir, version_dict):
+        """
+        update the dependency version of RackHD/debian/control
+        :param: debian_dir: the directory of RackHD/debian
+        :param: version_dict: a dictionay which includes the version of on-xxx
+        :return: None
+        """
+        control = os.path.join(debian_dir, "control")
+        print "1111111111111111111"
+        print control
+        print version_dict
+        if not os.path.isfile(control):
+            raise RuntimeError("Can't update dependency of {0} because it is not a file".format(control))
+
+        new_control = os.path.join(debian_dir, "control_new")
+        new_control_fp = open(new_control , "wb")
+
+        packages = self._get_control_depends(control)
+
+        with open(control) as fp:
+            package_count = 0
+            is_depends = False
+            for line in fp:
+                if line.startswith('Depends'):
+                    package_count += 1
+                    is_depends = True
+                    new_control_fp.write("Depends: ")
+                    # Start to write the dependes
+                    # If the depends is on-xxx, it will be replace with on-xxx (= 1.1...)
+                    for package in packages:
+                        package_name = package.split(',',)[0].strip()
+                        if ' ' in package_name:
+                            package_name = package_name.split(' ')[0]
+                        if package_name in version_dict:
+                            if ',' in package:
+                                depends_str = "         {0} (= {1}),{2}".format(package_name, version_dict[package_name], os.linesep)
+                            else:
+                                depends_str = "         {0} (= {1}){2}".format(package_name, version_dict[package_name], os.linesep)
+                            new_control_fp.write(depends_str)
+                        else:
+                            new_control_fp.write("{0}{1}".format(package, os.linesep))
+                else:
+                    if not is_depends or package_count >= len(packages):
+                        new_control_fp.write(line)
+                    else:
+                        package_count += 1
+
+        new_control_fp.close()
+        os.remove(control)
+        os.rename(new_control, control)
```

Comment:
The method is a static method,
I added try...except... outside of  calling the method.